_posts/2021-02-18-survey/survey.Rmd

```diff
@@ -0,0 +1,314 @@
+
+Thirdly, several respondents expressed the wish for more documentation, examples, and teaching materials. Let me say that here, the situation is different for TensorFlow than for `torch`.
+
+For [`tensorflow`](https://tensorflow.rstudio.com/), the website has a multitude of guides, tutorials, and examples. For [`torch`](torch.mlverse.org), reflecting the discrepancy in respective lifecycles, materials are not that abundant (yet). However, after a recent refactoring, the website has a new, four-part [Get started](https://torch.mlverse.org/start/) section addressed to both beginners in DL and experienced TensorFlow users curious to learn about `torch`. After this hands-on introduction, a good place to get more technical background would be the section on [tensors, autograd, and neural network modules](https://torch.mlverse.org/technical/).
+
+Truth be told, though, nothing would be more helpful here than contributions from the community. Whenever you solve even the tiniest (which is often how things appear to oneself) problem, consider creating a vignette explaining what you did. Future users will be thankful, and a growing user base means that over time, it'll be your turn to find that some things have already been solved for you!
+
+Which leads me to the last -- aggregate -- point.
+
+##### Community, community, community
+
+The remaining things I'd like to mention didn't come up quite as often (individually), but taken together, they all have something in common: They all are wishes *we* happen to have, as well!
+
+This definitely holds in the abstract -- let me cite:
+
+> "Develop more of a DL community"
+>
+> "Larger developer community and ecosystem. Rstudio has made great tools, but for applied work is has been hard to work against the momentum of working in Python."
+
+We wholeheartedly agree: And building a larger community is exactly what we're trying to do. I like the formulation "a DL community" insofar it is framework-independent. In the end, frameworks are just tools, and what counts is our ability to usefully apply those tools to problems we need to solve.
+
+Concrete wishes include
+
+-   More paper/model implementations (such as TabNet).
+
+-   Facilities for easy data reshaping and pre-processing (e.g., in order to pass data to RNNs or 1dd convnets in the expected 3-d format).
+
+-   Probabilistic programming for `torch` (analogously to TensorFlow Probability)
+
+-   A high-level library (such as fast.ai) based on `torch.`
+
+If time and resources were infinite, all of these would have become reality already.
+
+But then, instead of having everything implemented by a small group of people, isn't it much more effective (and satisfying!) to have a large and diverse group of people each contribute "theirs": the things they're interested in most, or which they feel drawn to most due to their individual backgrounds? In other words, if you're motivated to do so, please feel very encouraged to contribute to the growing community!
```

Comment:
I would change this from a question to a sentence. Our hope is that… and then end with "In other words…"

I don't know quite how to describe it, but you're asking something of people (to help out, which makes sense!), and don't want it to feel like you're doing this as a mere exercise (because it's more effective and satisfying) when, in reality, this is how open source works. Does that make sense?

Basically, we want to encourage people to help because we _need_ their help, and it feels a skosh paternalizing to say: _Wouldn't it be more fun and effective if we all chipped in?_

Reply:
hm... will reformulate ... (((will also clarify on slack)))

Reply:
changed to:
> In other words, there is a whole cosmos of useful things to create; and no small group alone can do it. This is where we hope we can build a community of people, each contributing what they're most interested in, and to whatever extent they wish.